crates/wasmtime/src/component/linker.rs

```diff
@@ -289,7 +326,7 @@ impl<T> LinkerInstance<'_, T> {
     /// argument which can be provided to the `func` given here.
     //
     // TODO: needs more words and examples
-    pub fn func_wrap<F, Params, Return>(&mut self, name: &str, func: F) -> Result<()>
+    pub fn func_wrap<F, Params, Return>(&mut self, name: &str, func: F) -> Result<PathIndex>
```

Comment:
I was thinking that the `PathIndex` here would only be returned for `fn resource` since it's not really applicable to any of the other types of imports. That'd also enable pushing `PathIndex` into the `Definition::Resource` variant instead of having it for all variants.